internal/app/query/query_service.go

```diff
@@ -135,7 +135,53 @@ func (s *queryService) QueryTicketIds(req *pb.QueryTicketIdsRequest, responseSer
 }
 
 func (s *queryService) QueryBackfills(req *pb.QueryBackfillsRequest, responseServer pb.QueryService_QueryBackfillsServer) error {
-	return status.Error(codes.Unimplemented, "not implemented")
+	if req == nil {
+		return status.Error(codes.InvalidArgument, "request entry cannot be nil")
+	}
+
+	pool := req.GetPool()
+	if pool == nil {
+		return status.Error(codes.InvalidArgument, ".pool is required")
+	}
+
+	pf, err := filter.NewPoolFilter(pool)
+	if err != nil {
+		return err
```

Comment:
Should it be an exact status code like in the return statements above? If so, please check other returns in this method.

Reply:
`QueryTickets` does not have. But we can make it better.

Reply:
I assume nothing to be done. `NewPoolFilter` already included those `codes`.

Reply:
Yes, but what should be done with other returns down below?

Reply:
The NewPoolFilter returns status.Error(...) if some error happens, so here that error it is only returned